### PR TITLE
DOC: interpolate: add a tutorial page for smoothing splines

### PR DESCRIPTION
**Pull request by ev-br**
Having some narrative docs for new additions gets more relevant as we plan to nudge users away from FITPACK (cf https://discuss.scientific-python.org/t/declare-univariatespline-classes-and-splrep-splprep-functions-legacy/1432).
Thus add some discussion, focusing mainly on high-level discussion of the math content of the several functions we provide and their basic usage patterns. Gory details are delegated to the API docs.

### CODE REVIEW

File: doc/source/tutorial/interpolate/smoothing_splines.rst [hunk 1]
```diff
@@ -6,53 +6,397 @@
...
+:math:`\mathbf{p}_j = (x_j, y_j)`, and we want to construct a parametric function
+:math:`\mathbf{g}(\mathbf{p}) = (g_x(u), g_y(u))`, where the values of the
+parameter :math:`u_j` correspond to :math:`x_j` and :math:`y_j`.
+
+Note that this problem readily generalizes to higher dimensions, :math:`d > 2`:
+we simply have :math:`d` data arrays and construct a parametric function with
+:math:`d` components.
+
+Also note that the choice of parametrization cannot be automated, and different
+parametrizations can lead to very different curves for the same data,
```
Comment (j-bowhay) [inline doc/source/tutorial/interpolate/smoothing_splines.rst:314]:
```suggestion
parameterizations can lead to very different curves for the same data,
```

**Review by j-bowhay — COMMENTED** [review]
A few minor comments but otherwise looks good

File: doc/source/tutorial/interpolate/smoothing_splines.rst [hunk 2]
```diff
@@ -6,53 +6,397 @@
...
+
+We provide two approaches to constructing smoothing splines, which differ in
+(1) the form of the penalty term, and (2) the basis in which the smoothing
+curve is constructed. Below we consider these two approaches.
+
+The former variant is performed by the `make_smoothing_spline` function,
+which is a clean-room reimplementation of the classic ``gcvspl`` Fortran package
+by H. Woltring.
+The latter variant is implemented by the `make_splrep` function, which is a
+reimplementation of the Fortran FITPACK library by P. Dierckx. :ref:`A legacy interface
```
[the remark this answers is not in the document]
Comment (ev-br) [reply doc/source/tutorial/interpolate/smoothing_splines.rst:31]: Yeah, when typing a text with TeX formulas the reflex is to add references :-). Here though, the API docs have them, cf https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.make_smoothing_spline.html and  https://scipy.github.io/devdocs/reference/generated/scipy.interpolate.make_splrep.html
So I'm two minds on whether to go all the way (then the tutorial will have quite a few) or refer to the API docs or...---WDYT?

[the remark this answers is not in the document]
Comment (j-bowhay) [reply doc/source/tutorial/interpolate/smoothing_splines.rst:31]: Ok that makes sense, we can always add them later if someone wants them

**Review by j-bowhay — APPROVED** [review]
One final typo but otherwise this looks great

**Comment by ev-br** [comment]
Thank you Jake for the review!

**Pull request merged**